dotnetapp-dev/Dockerfile

```diff
@@ -1,11 +1,35 @@
-FROM microsoft/dotnet:2.0-sdk
+FROM microsoft/dotnet:2.0-sdk AS test
 WORKDIR /app
 
 # copy csproj and restore as distinct layers
-COPY *.csproj ./
+COPY *.sln .
+COPY dotnetapp/*.csproj ./dotnetapp/
+COPY botutils/*.csproj ./botutils/
+COPY bottests/*.csproj ./bottests/
 RUN dotnet restore
 
 # copy and build everything else
-COPY . ./
-RUN dotnet publish -c Release -o out
-ENTRYPOINT ["dotnet", "out/dotnetapp.dll"]
+COPY dotnetapp/. ./dotnetapp/
+COPY botutils/. ./botutils/
+COPY bottests/. ./bottests/
+
+# run this stage for testing, with:
+# docker build --target test -t app:test .
+# docker run -ti -v C:\git\dotnet-docker-samples\dotnetapp-dev\TestResults:C:\app\bottests\TestResults app:test
+# docker run -ti -v C:\git\dotnet-docker-samples\dotnetapp-dev\TestResults:/app/bottests/TestResults app:test
+# After running, you will find test results on your local disk @ left-hand side of volume-mount syntax
+# First `docker run` above is for Windows containers. Second is for Linux.
```

Comment:
Second syntax really isn't for `Linux` - it is for a Windows machine running in Linux container mode.